src/borg/archiver.py

```diff
@@ -527,7 +529,8 @@ def _process(self, archive, cache, matcher, exclude_caches, exclude_if_present,
                                               read_special=read_special, dry_run=dry_run)
                         return
                 if not dry_run:
-                    status = archive.process_dir(path, st)
+                    if not recurse_excluded_dir:
+                        status = archive.process_dir(path, st)
```

Comment:
I don't claim to know lots about the details of borg's storage method, but I'm assuming that each backup operation stores file chunks in some manner that doesn't depend on order, but tracks the backup itself in the form of some manifest of the backed-up items.  If this is the case, then it could be solved as simply as just ensuring that the manifest is sorted before storing it such that parent folders come before their children.  This would work in conjunction with the idea of maintaining a list of parents to process after a matching nested folder is found.  Does this sound right or doable?